### PR TITLE
Fix extension configurations

### PR DESCRIPTION
**Pull request by CGNonofr**

### CODE REVIEW

File: update-extensions.ts [hunk 1]
```diff
@@ -126,18 +126,59 @@ function overrideDefaultValue (configuration: monaco.extra.IConfigurationNode) {
...
+        const message: string | undefined = typeof translated === 'string' || translated == null ? translated : (typeof translated.message === 'string' ? translated.message : undefined)
+        if (message !== undefined) {
+          obj[key] = message
+        } else {
+          console.warn(`Couldn't find message for key ${messageKey}.`)
+        }
+      }
+    } else if (isObject(value)) {
+      for (const k in value) {
+        if (Object.prototype.hasOwnProperty.call(value, k)) {
```
Comment (MaximeCheramy) [inline update-extensions.ts:195]: why this instead of `value.hasOwnProperty(k)`?

File: update-extensions.ts [hunk 2]
```diff
@@ -126,18 +126,59 @@ function overrideDefaultValue (configuration: monaco.extra.IConfigurationNode) {
...
-  } else if (typeof object === 'object') {
-    return Object.fromEntries(Object.entries(object).map(([key, value]) => [key, applyI18n(value, i18n)]))
-  } else {
-    return object
+interface MessageBag {
+  [key: string]: string | { message: string, comment: string[] } | undefined
+}
+
+function isString (value: unknown): value is string {
+  return Object.prototype.toString.call(value) === '[object String]'
```
Comment (MaximeCheramy) [inline update-extensions.ts:133]: please let me know if you remember why you had to do this instead of `typeof value === 'string'` :)

Comment (CGNonofr) [reply update-extensions.ts]: no reason a257a082fe20e4c3485f81db1b6ad3189c529b37

(It doesn't change the outputed files)

File: update-extensions.ts (hunk 1 above)
Comment (CGNonofr) [reply update-extensions.ts:198]: typescript error: `Do not access Object.prototype method 'hasOwnProperty' from target object.`

I quick search on the internets told me to use this syntax instead

Comment (CGNonofr) [reply update-extensions.ts:198]: https://eslint.org/docs/rules/no-prototype-builtins

**Review by MaximeCheramy — APPROVED** [review]

**Pull request merged**